### DISCUSSION TITLE
FastAccelStepperEngine as a static class member

### DISCUSSION THREAD

**Discussion by DreiDe** [gin66/FastAccelStepper#149] [Q&A] [answered]
I'm trying to use FastAccelStepper inside a custom stepper class. So far I could only get this working by initializing the engine globally and passing it to an init() method of my class that is called inside the setup function. Is it possible to "hide" the engine e.g. as a static member inside my stepper class? Or at least pass it as a constructor argument to the stepper class so no extra init method is needed?

**Comment by gin66** [comment, marked as answer]
Never tried this, but I assume, that the FastAccelStepperEngine instance could be stored anywhere, even inside your own class. Eventually you need to check, if the engine has been initialized and make sure, the engine is initialized only once.

**Reply by DreiDe** [reply]
Yes, you are probably right about this. I got it working now. The problem I was experiencing was caused by the constructor of my Stepper class being called before the `setup()` function. Thus SPI (for TMC2130) was not initialized and eventually also problems with `pinMode()` occurred.

Initializing the engine multiple times didn't cause any problems by the way.
